lib/simplifyExpression/fractionsSearch/cancelLikeTerms.js

```diff
@@ -20,6 +20,52 @@ class CancelOutStatus {
 // e.g. (2x^2 * 5) / 2x^2 => 5 / 1
 // Returns a Node.Status object
 function cancelLikeTerms(node) {
+  // Cancel out terms if the term being multiplied
+  // is the same as the denominator
+  if (Node.Type.isOperator(node) && node.op === '*') {
+    let newNode = clone(node);
+    let leftNode = newNode.args[0];
+    let rightNode = newNode.args[1];
```

Comment:
The code on master doesn't cancel out terms for `(2+x) * (3+x)/(2+x)`, which was what prompted these changes.  Currently it would distribute out the (2+x).  Without canceling, my changes caused an infinite loop by multiplying the denominator, distributing out, multiplying the denominator, etc.

Regarding the left node / right node positioning, I had made an assumption based on what would happen during equations steps, but you're right that it misses out on if an equation starts as `(2+x) * (3+x)/(2+x)`.  I'll work on resolving this in the next commit

From what I can tell that scenario wouldn't occur during equation solving though. Instead of multiplying by an inverted fraction, the steps would multiply by the denominator and then divide by the numerator